yt/utilities/lib/misc_utilities.pyx

```diff
@@ -1021,40 +1024,41 @@ def gravitational_binding_energy(
         np.float64_t[:] y,
         np.float64_t[:] z,
         int truncate,
-        np.float64_t kinetic):
+        np.float64_t kinetic,
+        int num_threads = 0):
 
     cdef int q_outer, q_inner, n_q, i
     cdef np.float64_t mass_o, x_o, y_o, z_o
     cdef np.float64_t mass_i, x_i, y_i, z_i
-    cdef np.float64_t this_potential, total_potential
-    total_potential = 0.
+    cdef np.float64_t total_potential = 0.
+    cdef np.float64_t this_potential
 
-    i = 0
     n_q = mass.size
-    pbar = get_pbar("Calculating potential for %d cells" % n_q,
-                    0.5 * (n_q * n_q - n_q))
-    for q_outer in range(n_q - 1):
+    print("Calculating potential for %d cells using %d threads" % (n_q,num_threads))
+
+    # using reversed iterator in order to make use of guided scheduling 
+    # (inner loop is getting more and more expensive)
+    for q_outer in prange(n_q - 1,-1,-1,
+        nogil=True,schedule='guided',num_threads=num_threads):
```

Comment:
Depending on how long this takes (e.g. anything more than a few seconds) you probably want to occasionally re-aquire the GIL and check for signals. This means that the calculation will be interruptable by, say, KeyboardInterrupt. Here's an example where we do that elsewhere in some cython code:

https://github.com/yt-project/yt/blob/5e1a06798e2cb694550352d0218e4ecda14e404f/yt/utilities/lib/image_samplers.pyx#L175-L177

It does add some overhead to reaquire the GIL and all threads must stop while the GIL is acquires, so make sure you're not doing it so often that it kills the parallelism but also make sure you're doing it often enough that a user doesn't need to wait more than half a second or so to kill the python interpreter after issuing `KeyboardInterrupt`.